### PR TITLE
Enhance Cluster Health Check

### PR DESCRIPTION
**Pull request by prsurve**
Fixes: #1821 

Signed-off-by: Pratik Surve <prsurve@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -144,6 +172,18 @@ def scan_cluster(self):
...
+        self.cephfsplugins = pod.get_plugin_pods(constants.CEPHFILESYSTEM, self.namespace)
+        self.rbdplugin_provisioners = pod.get_rbdfsplugin_provisioner_pods(
+            self.rbdplugin_provisioner_selector, self.namespace
+        )
+        self.cephfsplugin_provisioners = pod.get_cephfsplugin_provisioner_pods(
+            self.cephfsplugin_provisioner_selector, self.namespace
+        )
+        self.drain_canarys = pod.get_draincanary_pods(self.drain_canary_selector, self.namespace)
+
+        if config.ENV_DATA.get('platform') == constants.VSPHERE_PLATFORM:
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:185]: also BM needs to be checked here

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -650,6 +762,214 @@ def time_taken_to_complete_rebalance(self, timeout=600):
                 time_taken = time.time() - start_time
                 return (time_taken / 60)
 
+    def osd_health_check(self, count):
+        """
+        OSD health check based on pod count
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:767]: add an empty line between docstring sections

File: ocs_ci/ocs/cluster.py [hunk 3]
```diff
@@ -650,6 +762,214 @@ def time_taken_to_complete_rebalance(self, timeout=600):
...
+        Args:
+            count (int): Expected number of Osd pods
+        Raises:
+            OsdCountException: if osd pod count doesn't match
+        """
+        timeout = 10 * len(self.pods)
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:773]: we need to re-visit this method of calculation, as stated here by @mbukatov https://github.com/red-hat-storage/ocs-ci/pull/2287#discussion_r440932218

File: ocs_ci/ocs/cluster.py (hunk 1 above)
Comment (prsurve) [reply ocs_ci/ocs/cluster.py]: Done

File: ocs_ci/ocs/cluster.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/ocs/cluster.py:771]: Done

File: ocs_ci/ocs/cluster.py (hunk 3 above)
Comment (prsurve) [reply ocs_ci/ocs/cluster.py:779]: As expecting more time? because we are checking pod status, count after ceph health is OK

Comment (ebenahar) [reply ocs_ci/ocs/cluster.py:779]: @shylesh saw you initially added this calculation in https://github.com/red-hat-storage/ocs-ci/pull/362. Could you please elaborate on it?
@prsurve we can continue this discussion in an issue. Will not hold this PR

**Comment by petr-balogh** [comment]
Closing due to inactivity! Feel free to re-open or send the new PR.

**Pull request closed without merging**